### ISSUE TITLE
Technological choices

### ISSUE THREAD

**Issue by PBrunot** [fablab-bergamo/rfid-arduino#5]
The current implementation has an MQTT client only with MongoDB database. There is no MQTT broker for the  boards to connect.  In my opinion a"self-contained" python server with minimal external dependencies could be desiderable.

Protocols for the boards, options I see:
- MQTT Broker : [amqtt](https://github.com/Yakifo/amqtt)
- REST API sserer : https://flask.palletsprojects.com/en/2.2.x/ (to stay in line with FabManager architecture)

For database, options I see :
- MongoDB (current code base)
- PostgreSQL (to keep alignment with FabManager)
- SQLite (to get rid of independent DB engine, simplicity of development/installation)

My personal preference would be API REST + SQLite.

**Comment by valerionew** [comment]
I don't have any opinion on this. We may ask @lorossi and @MauroMombelli perhaps if they have anything to add

**Comment by PBrunot** [comment]
ok so for MQTT the libraries will be :
- esp32/adrduino : 256dpi/MQTT (MIT license). Client-side implemented and tested with mosquitto broker. It seems to work fine (so far). See https://github.com/fablab-bergamo/rfid-arduino/issues/6 

- python server side on Raspberry Pi : I will use [amqtt](https://github.com/Yakifo/amqtt) (MIT License)

**Comment by valerionew** [comment]
Why did you pick these two libraries instead of the more common [PubSubClient](https://github.com/knolleary/pubsubclient) for esp32 and [mosquitto](https://github.com/eclipse/mosquitto) for the server side?

The choice is fine, I'm just wondering if there is any reason to not use the two most widespread libraries/softwares

**Comment by PBrunot** [comment]
I actually used mosquitto on PC to test the ESP32 client library, it is easy to use. But I was hoping to have only one python program (without external programs) for server side, using Python libraries for Database, web server, MQTT broker. But maybe that's a wrong approach. I don't know if it's easy to "bundle & configure" things in Raspberry Pi.

For ESP32, I picked 256dpi/MQTT because it seems actively maintained (PubSubClient last commit is 3 years ago) and it was available in platformio repository. Licenses are the same (MIT). I can easily change to PubSubClient if this is a more common choice.
